### PR TITLE
tests: Make vlan and some bond tests nondestructive

### PR DESCRIPTION
**Pull request by martinpitt**
- [x] Builds on top of #13877
 - [x] Tests detect Machines iface detection bug: PR #13888
 - [x] Fix testNICAdd: PR #13899

### CODE REVIEW

**Comment by martinpitt** [comment]
Possibly relevant failure on [ubuntu-1804](https://logs.cockpit-project.org/logs/pull-13879-20200409-090258-dff7021b-ubuntu-1804/log.html#257), although this also just looks like slow DHCP.

**Comment by martinpitt** [comment]
Adjusted the tests, the naughty just covered another NM bug that affected ubuntu-1804.

**Comment by martinpitt** [comment]
@mvollmer: This is still blocked on #13877, but that one is approved and should be good. Do you mind having a look whether that general approach suits you? Then I can start working on the bridge tests (these will look very similar, I figure)

File: test/verify/check-networking-bond [hunk 1]
```diff
@@ -193,10 +198,55 @@ class TestNetworking(NetworkCase):
         b.click("#networking-interfaces tr[data-interface='tbond'] td:first-child")
         b.wait_visible("#network-interface")
         b.wait_present("#network-interface-slaves tr[data-interface='%s']" % iface)
-        b.wait_in_text("#network-interface .panel:contains('tbond')", ip)
+        # We want that to be true, but this only works for identical MACs (https://bugzilla.redhat.com/show_bug.cgi?id=1548265)
+        # b.wait_in_text("#network-interface .panel:contains('tbond')", ip)
+        # ... so just make sure that the bond gets some valid IP; this is broken on old NM as well
+        if self.machine.image not in ["ubuntu-1804"]:
```
Comment (martinpitt) [inline test/verify/check-networking-bond:204]: This special case can be dropped now, ubuntu-1804 just got dropped.

**Comment by martinpitt** [comment]
Interesting, this [triggers a bug](https://logs.cockpit-project.org/logs/pull-13879-20200409-231204-65f839e9-centos-8-stream/log.html#96-2) on the Machines page:

[screenshot](https://logs.cockpit-project.org/logs/pull-13879-20200409-231204-65f839e9-centos-8-stream/TestMachinesDBus-testNICAdd-centos-8-stream-127.0.0.2-2201-FAIL.png)

"Unable to get index for interface bonding_masters: No such device"

Apparently something is iterating over /sys/class/net/, and stumbling over this pseudo-interface? It appears as a regular file, not a symlink.

@KKoukiou

**Comment by martinpitt** [comment]
I sent PR #13888 to fix the network device detection, as it's an unrelated fix. But as that bug gets spotted here, adding blocked.

**Comment by martinpitt** [comment]
#13888 landed, thanks @skobyda for reviewing! Retrying tests.

**Comment by martinpitt** [comment]
Meh, now [testNICAdd](https://logs.cockpit-project.org/logs/pull-13879-20200413-123335-65f839e9-fedora-32/log.html#96-2) fails later in the game with an "interface busy" error.. so once again back to "blocked".

**Comment by martinpitt** [comment]
@mvollmer: This should finally be good to land now, I just retried the last unrelated flakes. Can you please review?

File: test/verify/check-networking-bond [hunk 2]
```diff
@@ -193,10 +198,54 @@ class TestNetworking(NetworkCase):
         b.click("#networking-interfaces tr[data-interface='tbond'] td:first-child")
         b.wait_visible("#network-interface")
         b.wait_present("#network-interface-slaves tr[data-interface='%s']" % iface)
-        b.wait_in_text("#network-interface .panel:contains('tbond')", ip)
+        # We want that to be true, but this only works for identical MACs (https://bugzilla.redhat.com/show_bug.cgi?id=1548265)
```
Comment (mvollmer) [inline test/verify/check-networking-bond:201]: This is the whole point of the test, no?  To ensure that bonding a active interface does not change the IP address of that interface.

**Comment by martinpitt** [comment]
@mvollmer: So what do you suggest? I veto keeping the status quo of eternally skipping the test that has never worked anywhere (https://github.com/cockpit-project/bots/issues/66). There is no open bugzilla for it, it's been that way for at least two years -- i. e. what's the real way out of this? Can we force the interface to get the exact same IP from DHCP through copying the MAC address first or so? How would one do that through the cockpit UI? Is that a realistic scenario even?

**Comment by mvollmer** [comment]
> So what do you suggest? 

Alright, this is confusing. TLDR: let's keep the "same IP address" requirement in testBondActive, and lets implement it by explicitly setting the MAC of the new bond.

Here is a time line of what likely happened, stretched out over a couple of years:

- We got the requirement that putting the main connection into a bond should not disconnect Cockpit. -> testBondingMain

- At the time, the main interface was using DHCP in our tests, so testBondingMain would be a realistic test. At the time a new bond made from a single active interface would indeed inherit the MAC of that interface, and the test would check that DHCP works on that interface.

- Then the main interface in our tests was changed to have a manual IP configuration.  Cockpit would copy that config when making a bond, so the bond created by testBondingMain would not use DHCP anymore and trivially have the right IP address.

- Then NM got the bug where it didn't properly enslave active interfaces, https://bugzilla.redhat.com/show_bug.cgi?id=1548265. testBondingMain should have caught this because the bond didn't actually work anymore and DHCP would not succeed, but since it now used a manual config, we didn't catch that.

- So I made testBondActive to catch bug 1548265, and it did.  We made a naugthy for it, with the expecation that NM will fix this eventually.  However, I also wrote the assumption into the test that the bond will get the same IP as the enslaved interface, and we were conflating "bond gets an address from DHCP" with "bond gets the same IP as the interface".  The naughty pattern should only caught the first, but it also caught the latter by accident.

- NM/kernel/whatever changed behavior with bond creation and now bonds don't inherit the MAC from the first interface anymore.  testBondingMain was not affected because of its manual config. testBondActive was still failing to properly enslave the interface.

- NM fixed bug 1548265 and the bond in testBondActive got an IP address, but it was not the same as the interfaces.  The naughty pattern still matched, and we missed that.

So the old requirement that a bond uses the MAC of its first slave seems to have disappeared. It's not happening anymore and nobody seems to care.

Cockpit now lets you explicitly select the MAC of a bond, and that's probably what people do that need a bond to inherit the IP of a specific interface.

[ I didn't really do any research about the timeline above. It migth be totally bogus. ]

**Comment by mvollmer** [comment]
> I veto keeping the status quo of eternally skipping the test that has never worked anywhere (cockpit-project/bots#66).

testBondActive had succeed on rhel-7-4 at the time: https://bugzilla.redhat.com/show_bug.cgi?id=1548265#c11

**Comment by mvollmer** [comment]
I'll push a FIXUP.

**Comment by martinpitt** [comment]
Thanks @mvollmer: Tests look good, so I pushed a squashing with no effective code changes. Can you please re-review?

**Pull request merged**